Sources/_StringProcessing/PrintAsPattern.swift

```diff
@@ -578,7 +578,7 @@ extension PrettyPrinter {
 extension String {
   // TODO: Escaping?
   fileprivate var _quoted: String {
-    "\"\(self._replacing("\"", with: "\\\""))\""
+    "\"\(self._replacing("\\", with: "\\\\")._replacing("\"", with: "\\\""))\""
```

Comment:
```suggestion
    "\"\(self._replacing(#"\"#, with: #"\\"#)._replacing(#"""#, with: #"\""#))\""
```

Reply:
Or we can emit it as a raw string if it has any backslashes or quotes inside it. We could do a quick check to count the number of hashes prior to a backslash and emit a raw string with one additional one.